File: tradescout-api/src/Invoice/Invoice.entity.ts
import { Entity, PrimaryGeneratedColumn, Column, ManyToOne, OneToMany, JoinColumn, CreateDateColumn, UpdateDateColumn, DeleteDateColumn } from 'typeorm';
import { InvoiceStatusEnum } from './InvoiceEnums';
import { Business } from '../Business/Business.entity';
import { Expense } from '../Expense/Expense.entity';
import { Income } from '../Income/Income.entity';

@Entity('invoices')
export class Invoice {
  @PrimaryGeneratedColumn({type: 'integer'})
  id: number;

  @Column()
  invoiceNumber: string;

  @Column()
  customerName: string;

  @Column('decimal', { precision: 10, scale: 2 })
  totalAmount: number;

  @Column({ enum: InvoiceStatusEnum, default: 'DRAFT' })
  status: InvoiceStatusEnum;

  @Column({type: 'jsonb', nullable: true})
  snapshotData?: Record<string, any>;

  @Column({ type: 'timestamp', nullable: true })
  issuedAt?: Date;

  @CreateDateColumn()
  createdAt: Date;

  @UpdateDateColumn()
  updatedAt: Date;

  @DeleteDateColumn()
  deletedAt: Date



  // --- Relationships ---
  
  @ManyToOne(() => Business, (business) => business.invoices)
  @JoinColumn({ name: 'businessId' })
  business: Business;

  @Column()
  businessId: string;

  @OneToMany(() => Income, (income) => income.invoice)
  income: Income[];

  @OneToMany(() => Expense, (expense) => expense.invoice)
  expenses: Expense[];
}

File: api/Entities/InvoiceEntity.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json;

namespace api.Entites
{
    public enum InvoiceStatusEnum
    {
        DRAFT,
        PAID,
        UNPAID,
        PARTIAL,
        VOID
    }
    [Table("invoices")]
    public class Invoice
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public string InvoiceNumber { get; set; } = string.Empty;

        [Required]
        public string CustomerName { get; set; } = string.Empty;

        [Column(TypeName = "decimal(10, 2)")]
        public decimal TotalAmount { get; set; }

        public InvoiceStatusEnum Status { get; set; } = InvoiceStatusEnum.Draft;

        [Column(TypeName = "jsonb")]
        public JsonDocument? SnapshotData { get; set; }

        public DateTime? IssuedAt { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime UpdatedAt { get; set; }

        public DateTime? DeletedAt { get; set; }



        // [Required]
        // public string BusinessId { get; set; } = string.Empty;
        // [ForeignKey(nameof(BusinessId))]
        // public Business? Business { get; set; }
        // public ICollection<Income> Payments { get; set; } = new List<Income>();
        // public ICollection<Expense> JobExpenses { get; set; } = new List<Expense>();
    }
}